### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by Wineshuga**
**Description**

In this branch, I completed the following steps:

- Added Redux Toolkit to the project using `npm install react-redux @reduxjs/toolkit`.
- Created a directory that will contain all your Redux logic `/src/redux`
- Configured a Redux store `/src/redux/store.js`
- Defined a slice of state for books that: 
    - [ ] Includes an array of books (initial state: empty array)
    - [ ] Includes a reducer that adds a book
    - [ ] Includes a reducer that removes a book
- Define a slice of state for categories that: 
    - [ ] Includes an array of categories (initial state: empty array)
    - [ ] Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const booksSlice = createSlice({
+  name: 'book',
+  initialState: [],
```
Comment (Cabraham1) [inline src/redux/books/booksSlice.js]: - [ ] Nice work so far in creating your initialState, but as per the project you are to have an empty book array
 
```
initialState: {
    books: [],
  },
```

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  reducers: {
+    addBook: (state) => [...state, { title: 'New Book' }],
```
Comment (Cabraham1) [inline src/redux/books/booksSlice.js]: - [ ] Nice work 👍🏿 so far but as per project requirements you are to get the title and author as payload  from action creators.
 please implement it 
 
 
 ```javascript
addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
```

File: src/redux/categories/categoriesSlice.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
```
Comment (Cabraham1) [inline src/redux/categories/categoriesSlice.js:9]: - [ ]  Nice work so far in creating your initialState, but as per the project you are to have an empty categories array

```javascript
initialState: {
    categories: [],
  },
```

**Review by Cabraham1 — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @Wineshuga   :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary


Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:


# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers, and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Wineshuga👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with redux. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Gitflow is followed. :heavy_check_mark:



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**